src/quicr_client_raw_session.h

```diff
@@ -0,0 +1,251 @@
+   *                                Request
+   */
+  void unsubscribe(const quicr::Namespace& quicr_namespace,
+                   const std::string& origin_url,
+                   const std::string& auth_token) override;
+
+  /**
+   * @brief Publish Named object
+   *
+   * @param quicr_name               : Identifies the QUICR Name for the object
+   * @param priority                 : Identifies the relative priority of the
+   *                                   current object
+   * @param expiry_age_ms            : Time hint for the object to be in cache
+   *                                      before being purged after reception
+   * @param use_reliable_transport   : Indicates the preference for the object's
+   *                                   transport, if forwarded.
+   * @param data                     : Opaque payload
+   *
+   */
+  void publishNamedObject(const quicr::Name& quicr_name,
+                          uint8_t priority,
+                          uint16_t expiry_age_ms,
+                          bool use_reliable_transport,
+                          bytes&& data) override;
+
+  /**
+   * @brief Publish Named object
+   *
+   * @param quicr_name               : Identifies the QUICR Name for the object
+   * @param priority                 : Identifies the relative priority of the
+   *                                   current object
+   * @param expiry_age_ms            : Time hint for the object to be in cache
+                                       before being purged after reception
+   * @param use_reliable_transport   : Indicates the preference for the object's
+   *                                   transport, if forwarded.
+   * @param offset                   : Current fragment offset
+   * @param is_last_fragment         : Indicates if the current fragment is the
+   * @param data                     : Opaque payload of the fragment
+   */
+  void publishNamedObjectFragment(const quicr::Name& quicr_name,
+                                  uint8_t priority,
+                                  uint16_t expiry_age_ms,
+                                  bool use_reliable_transport,
+                                  const uint64_t& offset,
+                                  bool is_last_fragment,
+                                  bytes&& data) override;
+
+  void handle(messages::MessageBuffer&& msg);
+  void removeSubscribeState(bool all, const quicr::Namespace& quicr_namespace,
+                            const SubscribeResult::SubscribeStatus& reason);
+
+  std::shared_ptr<qtransport::ITransport> transport;
+  qtransport::LogHandler& log_handler;
+
+protected:
+  std::mutex session_mutex;
+
+  bool notify_pub_fragment(const messages::PublishDatagram& datagram,
+                           const std::map<int, bytes>& frag_map);
+  void handle_pub_fragment(messages::PublishDatagram&& datagram);
+
+  qtransport::LogHandler def_log_handler;
+
```

Comment:
The members are a little buried in here, could these be moved further down past the function and type declarations?

Reply:
Agreed. This is how the code was.  I didn't try to improve it. I just restructured so I can insert a different session.  This (among other things) needs to be revisited.